### ISSUE TITLE
iOS: Input field not working properly if  "-webkit-user-select: none" is used

### ISSUE THREAD

**Issue by duxthefux** [select2/select2#4689]
## Prerequisites

- [x ] I have searched for similar issues in both open and closed tickets and cannot find a duplicate
- [x ] The issue still exists against the latest `master` branch of Select2
- [x ] This is not a usage question (Those should be directed to the [community](https://select2.github.io/community.html))
- [x ] I have attempted to find the simplest possible steps to reproduce the issue
- [ ] I have included a failing test as a pull request (Optional)

## Steps to reproduce the issue

1. Set the following CSS Rule to disable user text selection
* {
    -webkit-touch-callout: none !important;
    -webkit-user-select: none !important;
    -webkit-tap-highlight-color: transparent !important;
    -moz-user-select: none !important;
}

2. Open safari on any ios device and try to type in the search field (not possible and if possible only for one letter)
3. You even need to tap the <input> field again to type in the field (not possible and if possible only for one letter)

## Expected behavior and actual behavior

When I follow those steps, I can't type in the input field at all or only one letter.

I was expecting normal search behaviour.

The fix for this issue is (didn't try to optimize the selectors)

.select2-container--default *{
    -webkit-user-select: text !important;
    -moz-user-select: text !important;
}

## Environment

Browsers

- [ ] Google Chrome
- [ x] Safari
- [ ] Mozilla Firefox
- [ ] Internet Explorer

Operating System

- [ ] Windows
- [ ] Mac OS X
- [x] iOS
- [ ] Linux
- [ ] Mobile

Libraries

- jQuery version: 1.11.1
- Select2 version: 4.0.3

## Isolating the problem

- [ ] This bug happens [on the examples page](https://select2.github.io/examples.html)
- [ ] The bug happens consistently across all tested browsers
- [x] This bug happens when using Select2 without other pluigns
- [ ] I can reproduce this bug in [a jsbin](https://jsbin.com/)

**Comment by ViroDaHaxr** [comment]
Had the SAME issue. Spent 2 hours working on it and was TOTALLY blown away and frustrated finding out it was a WebKit bug. 2 hours paid off though. Wish I found this github before spending 2 hours figuring out the issue. Anyway I second this guys claims of a bug.

**Comment by netrolite** [comment]
Had the same issue!
